### ISSUE TITLE
error when run th train.lua

### ISSUE THREAD

**Issue by tongerpeng** [facebookresearch/deepmask#47]
-- ignore option rundir	
-- ignore option dm	
-- ignore option reload	
-- ignore option gpu	
-- ignore option datadir	
| running in directory /home/tong/Downloads/deepmask-master/exps/deepmask/exp	
| number of paramaters trunk: 15213312	
| number of paramaters mask branch: 1608768	
| number of paramaters score branch: 526337	
| number of paramaters total: 17348417	
| start training	
/home/tong/torch/install/bin/luajit: /home/tong/torch/install/share/lua/5.1/threads/threads.lua:183: [thread 2 callback] bad argument #2 to '?' (out of range at /home/tong/torch/pkg/torch/generic/Tensor.c:890)
stack traceback:
	[C]: at 0x7f8621d976b0
	[C]: in function '__index'
	/home/tong/Downloads/deepmask-master/DataSampler.lua:135: in function 'maskSampling'
	/home/tong/Downloads/deepmask-master/DataSampler.lua:111: in function 'get'
	/home/tong/Downloads/deepmask-master/DataLoader.lua:70: in function </home/tong/Downloads/deepmask-master/DataLoader.lua:64>
	[C]: in function 'xpcall'
	/home/tong/torch/install/share/lua/5.1/threads/threads.lua:234: in function 'callback'
	/home/tong/torch/install/share/lua/5.1/threads/queue.lua:65: in function </home/tong/torch/install/share/lua/5.1/threads/queue.lua:41>
	[C]: in function 'pcall'
	/home/tong/torch/install/share/lua/5.1/threads/queue.lua:40: in function 'dojob'
	[string "  local Queue = require 'threads.queue'..."]:15: in main chunk
stack traceback:
	[C]: in function 'error'
	/home/tong/torch/install/share/lua/5.1/threads/threads.lua:183: in function 'dojob'
	/home/tong/Downloads/deepmask-master/DataLoader.lua:95: in function '(for generator)'
	/home/tong/Downloads/deepmask-master/TrainerDeepMask.lua:68: in function 'train'
	train.lua:117: in main chunk
	[C]: in function 'dofile'
	...tong/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
	[C]: at 0x00405d50

can anyone tell me how to fix it...

**Comment by PavlosMelissinos** [comment]
Copied from #72:

Line 135 in DataSampler (function maskSampling) is the following:
> local catId = self.catIds[cat]

The error is:
> (out of range at /home/zelle/torch/pkg/torch/generic/Tensor.c:890)

Print the size of the array (#self.catIds) right before that line, as well as cat, and you'll see that the former is smaller than the latter.
The problem is that the number of categories is hardcoded to be 80, because the code has only been tested on MS-COCO. Your dataset apparently has fewer classes than that; therefore you have to change line 132:
> local cat,ann = torch.random(80)

to represent the number of categories in your dataset.
